Linking Error

### Describe the bug

After pulling the latest develop, I encounter the following message.

In build, I used
`cmake .. -D USE_ELPA=OFF -D ENABLE_LIBRI=OFF -D BUILD_TESTING=OFF`
and 
`make`

```
[100%] Linking CXX executable abacus
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_dispatch_next_4'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_end_masked'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_for_static_fini'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_omp_task_alloc'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_dispatch_next_8'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_barrier'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_omp_task'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_dispatch_init_4'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_end_serialized_parallel'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_atomic_fixed4_add'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_dispatch_fini_8'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_for_static_init_4u'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_atomic_fixed8_add'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_atomic_float4_add'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_atomic_float8_max'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_reduce'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_end_critical'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_for_static_init_8'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_atomic_float8_add'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_end_reduce'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_flush'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_for_static_init_8u'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_fork_call'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_for_static_init_4'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_ordered'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_push_num_threads'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_atomic_cmplx8_add'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_global_thread_num'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_critical'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_omp_task_with_deps'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_reduce_nowait'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_atomic_float4_max'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_single'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_dispatch_init_8'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_atomic_cmplx4_add'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_ok_to_fork'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_end_single'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_end_ordered'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_end_reduce_nowait'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_serialized_parallel'
/usr/bin/ld: /opt/intel/oneapi/mkl/2025.0/lib/libmkl_intel_thread.so: undefined reference to `__kmpc_masked'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/abacus.dir/build.make:1300: abacus] Error 1
make[1]: *** [CMakeFiles/Makefile2:944: CMakeFiles/abacus.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```
It was fine before.

### Expected behavior

_No response_

### To Reproduce

_No response_

### Environment

_No response_

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

--- a/cmake/FindCereal.cmake
+++ b/cmake/FindCereal.cmake
@@ -15,7 +15,7 @@ if(NOT CEREAL_INCLUDE_DIR)
     include(FetchContent)
     FetchContent_Declare(
         cereal
-        URL https://codeload.github.com/USCiLab/cereal/tar.gz/master
+        URL https://codeload.github.com/MCresearch/cereal/tar.gz/master
     )
     FetchContent_Populate(cereal)
     set(CEREAL_INCLUDE_DIR ${cereal_SOURCE_DIR}/include)

